sdk/storage/azure-storage-quickquery/src/main/java/com/azure/storage/quickquery/implementation/util/FluxInputStream.java

```diff
@@ -0,0 +1,232 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.quickquery.implementation.util;
+
+import com.azure.core.util.FluxUtil;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.storage.blob.models.BlobStorageException;
+import com.azure.storage.common.implementation.Constants;
+import org.reactivestreams.Subscription;
+import reactor.core.publisher.Flux;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+import java.util.concurrent.locks.Condition;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+/**
+ * An InputStream interface that subscribes to a Flux allows data to be read.
+ */
+public class FluxInputStream extends InputStream {
```

Comment:
This thing doesn't seem to be qq/avro specific. Should this go to common?

Reply:
moving it to common